client/utils/checkout.js

```diff
@@ -30,10 +24,18 @@ export const getConfig = ( name ) => {
  */
 export const getUPEConfig = ( name ) => {
 	// Classic checkout or blocks-based one.
-	const config =
-		typeof wcpay_upe_config !== 'undefined'
-			? wcpay_upe_config
-			: wc.wcSettings.getSetting( 'woocommerce_payments_data' );
+	let config = null;
+	if ( typeof wcpay_upe_config !== 'undefined' ) {
+		config = wcpay_upe_config;
+	} else if (
+		typeof wc === 'object' &&
+		typeof wc.wcSettings !== 'undefined'
+	) {
+		// If getSettings or woocommerce_payments_data is not available, default to an empty object so we return null bellow.
+		config = wc.wcSettings?.getSetting( 'woocommerce_payments_data' ) || {};
```

Comment:
```suggestion
		config = wc.wcSettings.getSetting( 'woocommerce_payments_data' ) || {};
```

Since we're already checking above `typeof wc.wcSettings !== 'undefined'`, is the optional chaining operator still necessary?